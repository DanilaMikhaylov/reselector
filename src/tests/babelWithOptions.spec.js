import { transformFileSync } from '@babel/core'

let cache = {}

const setHash = ({ filename, id, name }) => {
  cache[id] = `${filename.split('/').slice(-2).join('/')} ${name}`
}

const plugins = [
  ['./src/babel', { setHash }],
]

const presets = [
  '@babel/preset-react',
]

describe('babel plugin with options', () => {
  let config = {}

  beforeEach(() => {
    config = {}
    cache = {}

    jest.doMock('../config', () => new Proxy(jest.requireActual('../config'), {
      get: (target, value) => (value in config ? config[value] : target[value]),
    }))
  })

  it('should cache all Elements', () => {
    transformFileSync(require.resolve('./App'), {
      plugins,
      presets,
    })

    expect(cache).toMatchSnapshot()
  })
})
