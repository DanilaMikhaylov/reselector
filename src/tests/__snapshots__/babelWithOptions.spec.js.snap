// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel plugin with options should cache all Elements 1`] = `
Object {
  "5613b2e5": "App/index.js ComponentNoDestructure",
  "5f9b51d4": "App/index.js default",
  "60d032ee": "App/index.js ComposedComponent2",
  "6da2cdef": "App/index.js SpreadPropsComponent",
  "889909b2": "App/index.js ComponentWithAttrs",
  "8e388bfc": "App/index.js ComposedComponent",
  "a8f769be": "App/index.js ClassComponent",
  "b5d7faff": "App/index.js FunctionalComponent",
  "c7b7156f": "App/index.js UnexportComponent",
  "ca715737": "App/index.js RenderFunction",
  "dda89e29": "App/index.js Button",
  "f49764c6": "App/index.js ArrowFunctionalComponent",
}
`;
